modules/http/src/main/scala/ch/epfl/bluebrain/nexus/service/http/routes/StaticResourceRoutes.scala

```diff
@@ -1,70 +1,57 @@
 import io.circe.parser.parse
 
 import scala.io.Source
 
 /**
   * Routes that expose static resources provided on the classpath
-  * @param resourcePath path to the folder in classpath to expose
+  * @param resourcePaths [[Map]] containing mapping between path at which resource will be available(starting with `/`)
+  *                     and the path where the resource can be found on the classpath.
   * @param prefix       prefix to prepend to the routes
   * @param baseUri      base URI to use in IDs, will replace {{base}} in all the resources
   */
-class StaticResourceRoutes(resourcePath: String, prefix: String, baseUri: Uri) {
+class StaticResourceRoutes(resourcePaths: Map[String, String], prefix: String, baseUri: Uri) extends PrefixDirectives {
 
-  private def contentOf(file: File): String =
-    Source.fromFile(file).mkString
+  private def contentOf(file: String): String =
+    Source.fromInputStream(getClass.getResourceAsStream(file)).mkString
 
-  private def contentOf(file: File, replacements: Map[String, String]): String =
+  private def contentOf(file: String, replacements: Map[String, String]): String =
     replacements.foldLeft(contentOf(file)) {
       case (value, (regex, replacement)) => value.replaceAll(regex, replacement)
     }
 
   private val baseReplacement: Map[String, String] = Map(quote("{{base}}") -> baseUri.toString)
 
-  private def folderContents(folder: File): Map[String, Json] = {
-    folder
-      .listFiles()
-      .toList
-      .filter(f => f.isFile && f.getName.endsWith(".json"))
-      .flatMap { file =>
-        parse(contentOf(file, baseReplacement)).toOption.map { json =>
-          file.getName.stripSuffix(".json") -> json
-        }
+  private lazy val resources: Map[String, Json] =
+    resourcePaths
+      .mapValues { resource =>
+        parse(contentOf(resource, baseReplacement)).toOption
+      }
+      .flatMap {
+        case (key, value) =>
+          value match {
+            case Some(v) => Some((key, v))
+            case None    => None
+          }
       }
-      .toMap
-  }
-
-  private lazy val resources: Map[String, Map[String, Json]] = {
-    val resourceFolder = new File(getClass.getResource(resourcePath).getPath)
-    if (resourceFolder.exists && resourceFolder.isDirectory) {
-      resourceFolder
-        .listFiles()
-        .toList
-        .filter(_.isDirectory)
-        .map { folder =>
-          folder.getName -> folderContents(folder)
-        }
-        .toMap
-    } else {
-      Map()
-    }
-  }
 
   def routes: Route =
-    (get & pathPrefix(prefix)) {
-      path(Segment / Segment) { (resourceType, resource) =>
-        resources.get(resourceType).flatMap(_.get(resource)) match {
-          case Some(json) => complete(json)
-          case None       => complete(StatusCodes.NotFound)
+    uriPrefix(baseUri) {
+      (get & pathPrefix(prefix)) {
+        extractUnmatchedPath { resourcePath =>
+          resources.get(resourcePath.toString) match {
```

Comment:
I think we should reject the route if we don't have a match, such that runtime routes continue to be tried after failure to match.